scripts/set-license-text.sh

```diff
@@ -0,0 +1,20 @@
+#!/usr/bin/env bash
+
+# This script tries to figure out what year a package was created and updates
+# its `LICENSE` file accordingly (including copyright owner)
+
+set -euo pipefail
+
+declare -A years
+for pkg in ./packages/*; do
+  year=$(git --no-pager log --diff-filter=A --follow --format=%ad --date format:%Y --reverse -- "${pkg}" | head -n1) || true
```

Comment:
`|| true` due to `pipefail`. `pipefail` causes the script to fail _but_ doesn't give me any information about what happened?

Reply:
I’m fine with this. I assume `head` exiting is causing `git` to write to a closed file. I’m not sure there’s a clean solution for this except to unnecessarily absorb the whole log into a buffer. I like using `git hash-object -w --stdin` for slurping, but I believe there’s something you can pipe inline. The buffering conceit presumes that `git cat-file blob $HASH` exits gracefully if it fails to write to stdout, unlike `git log`.